api/database/delegations.go

```diff
@@ -5,16 +5,27 @@ import (
 	"github.com/jmoiron/sqlx"
 )
 
-func (d *Database) Delegations(address string) ([]tracelistener.DelegationRow, error) {
-	var delegations []tracelistener.DelegationRow
+// DelegationResponse represents a delegation response got from database.
+type DelegationResponse struct {
+	tracelistener.DelegationRow
+
+	ValidatorTokens string `db:"tokens" json:"tokens"`
+	ValidatorShares string `db:"delegator_shares" json:"delegator_shares"`
+}
+
+func (d *Database) Delegations(address string) ([]DelegationResponse, error) {
+	var delegations []DelegationResponse
 
 	q, args, err := sqlx.In(`
-	SELECT * FROM tracelistener.delegations
-	WHERE delegator_address IN (?)
-	AND chain_name IN (
+	SELECT d.chain_name, d.delegator_address, d.validator_address, d.amount, v.tokens, v.delegator_shares
+	FROM tracelistener.delegations as d
+	INNER JOIN tracelistener.validators as v ON 
+		d.validator_address=v.validator_address
+	WHERE d.delegator_address IN (?)
+	AND d.chain_name IN (
 		SELECT chain_name FROM cns.chains WHERE enabled=true
 	)
-	AND delete_height IS NULL
+	AND d.delete_height IS NULL
```

Comment:
I know it's unlikely this will happen but to be safe:
```suggestion
	AND d.delete_height IS NULL
	AND v.delete_height IS NULL
```